cpp_linter/rest_api/github_api.py

```diff
@@ -446,6 +450,51 @@ def post_review(
                 summary_only=summary_only,
                 review_comments=review_comments,
             )
+        ignored_reviews = []
+        if not summary_only:
+            found_threads = self._get_existing_review_comments(
+                no_dismissed=reuse_comments and not delete_comments)
+            if found_threads:
+                if reuse_comments:
+                    # Keep already posted comments if they match new ones
+                    review_comments_suggestions = review_comments.suggestions
+                    review_comments.suggestions = []
+                    existing_review_comments = []
+                    for thread in found_threads:
+                        for comment in thread["comments"]["nodes"]:
+                            found = False
+                            line_start = comment["originalStartLine"] if comment["originalStartLine"] is not None else -1
+                            line_end = comment["originalLine"]
+                            if comment["startLine"] is not None:
+                                line_start = comment["startLine"] if comment["startLine"] is not None else (
+                                    comment["line"] if comment["line"] is not None else line_start)
+                                line_end = comment["line"] if comment["line"] is not None else line_end
+                            for suggestion in review_comments_suggestions:
+                                if (suggestion.file_name == comment["path"] and suggestion.line_start == line_start and suggestion.line_end == line_end and suggestion.comment == comment["body"] and suggestion not in existing_review_comments and thread["isResolved"] is False and thread["isCollapsed"] is False):
```

Comment:
Yikes! This is why we use ruff or black to format code:
```suggestion
                                if (
                                    suggestion.file_name == comment["path"]
                                    and suggestion.line_start == line_start
                                    and suggestion.line_end == line_end
                                    and suggestion.comment == comment["body"]
                                    and suggestion not in existing_review_comments
                                    and thread["isResolved"] is False
                                    and thread["isCollapsed"] is False
                                ):
```

Reply:
I used pep8 for formatting, ive updated it again using ruff

Reply:
we have pre-commit setup for this repo also.